plugin/fireplace.vim

```diff
@@ -1666,6 +1666,62 @@ augroup fireplace_go_to_file
   autocmd FileType clojure call s:set_up_go_to_file()
 augroup END
 
+" Section: Spec
+" TODO: `spec-list`
+
+function! fireplace#symbol_or_fallback(...) abort
+  if empty(a:1)
+    return expand('<cword>')
+  endif
+  return a:1
+endfunction
+
+function! fireplace#fully_qualified_symbol(symbol) abort
+  if a:symbol =~# '^::.\+/'
+    let symbol = ':' . fireplace#resolve_alias(matchstr(a:symbol, '^::\zs[^/]\+')) . matchstr(a:symbol, '/.*')
+  elseif a:symbol =~# '^::'
+    let symbol = ':' . fireplace#ns() . '/' . strpart(a:symbol, 2)
+  else
+    let symbol = a:symbol
+  endif
+  return symbol
+endfunction
+
+function! s:SpecForm(...) abort
+  let op = "spec-form"
+  if fireplace#op_available(op)
+    let symbol = fireplace#fully_qualified_symbol(fireplace#symbol_or_fallback(a:1))
+    let response = fireplace#message({'op': op, 'spec-name': symbol})[0]
+    if !empty(get(response, op))
+      " FIXME: needs to be pretty printed
+      echo get(response, op)
+    endif
+  endif
+  return ''
+endfunction
+
+function! s:SpecExample(...) abort
+  let op = "spec-example"
+  if fireplace#op_available(op)
+    let symbol = fireplace#fully_qualified_symbol(fireplace#symbol_or_fallback(a:1))
+    let response = fireplace#message({'op': op, 'spec-name': symbol})[0]
+    if !empty(get(response, op))
+      echo get(response, op)
+    endif
+  endif
+  return ''
+endfunction
+
+function! s:set_up_fireplace_spec() abort
+  command! -buffer -bar -nargs=? -complete=customlist,fireplace#eval_complete SpecForm    :exe s:SpecForm(<q-args>)
```

Comment:
Let's leave this as `-nargs=1` for now. I won't rule out the `<cword>` default but we should do it everywhere it makes sense if so.